### ISSUE TITLE
jsonschemagen should include format restrictions for dates 

### ISSUE THREAD

**Issue by cmungall** [linkml/linkml#260]
The kitchen sink test emits json like:

```
     "EmploymentEvent": {
         "additionalProperties": false,
         "description": "",
         "properties": {
            "employed_at": {
               "type": "string"
            },
            "ended_at_time": {
               "type": "string"
            },
            "is_current": {
               "type": "boolean"
            },
            "started_at_time": {
               "type": "string"
            }
         },
```

"format": "date" should be added if. the linkml type is date

see https://opis.io/json-schema/2.x/formats.html

**Comment by cmungall** [comment]
as per https://json-schema.org/understanding-json-schema/reference/string.html#format

> JSON Schema implementations are not required to implement this part of the specification, and many of them do not."
  
it appears the python json schema implementation does not implement this, so we skip the test for this for now

**Comment by cmungall** [comment]
Fixed by #261
